Package.swift

```diff
@@ -26,7 +26,9 @@ func resolveTargets() -> [Target] {
                 exclude: [infoPlist],
                 sources: ["Purchases"],
                 publicHeadersPath: "Purchases/Public",
-                cSettings: objcSources.map { CSetting.headerSearchPath($0) }
+                cSettings: objcSources.map { CSetting.headerSearchPath($0) } + [
+                    .define("NS_BLOCK_ASSERTIONS", to: "1", .when(configuration: .release))
```

Comment:
I'm in favor of this to have consistency across the different types of integration, but I'll let the team chime in.
If we merge this, could you add a comment linking to https://forums.swift.org/t/assertions-in-swift-packages/42692 for future reference?